### ISSUE TITLE
Configurable timeout setting

### ISSUE THREAD

**Issue by bra-fsn** [crossbario/txaio-etcd#4]
Hi,

When I run this code:

```python
try:
  r=yield etcd.set(b'key1', b'foobar')
  print r
except Exception:
  traceback.print_exc()
```
with a server, which is SIGSTOP-ped (so I can open a TCP connection to it, but it won't produce any answer), the program never gets through etcd.set, it doesn't time out. (at least not for some minutes)
lt would be good to have a configurable timeout.

**Comment by oberstet** [comment]
Should be straight forward using the `timeout` parameter of treq ([here](https://treq.readthedocs.io/en/latest/api.html#treq.request).

So we need an additional `timeout` keyword parameter in basically all public methods like `txaioetcd.Client.get`.

**Comment by oberstet** [comment]
implemented in v0.2.0
